core/lib/dal/migrations/20240612152419_create_l1_transactions_status.up.sql

```diff
@@ -0,0 +1,25 @@
+CREATE TYPE transaction_status AS ENUM ('Created', 'Pending', 'Confirmed', 'Failed');
+CREATE TYPE transaction_type AS ENUM ('Commit', 'PublishProofOnchain', 'Execute');
+
+ALTER TABLE eth_txs ADD COLUMN idempotency_key VARCHAR UNIQUE;
+
+CREATE TABLE l1_transactions
+(
+    -- uuid with some human readable prefix, acts as idempotency_key for eth-sender
+    id               VARCHAR            NOT NULL PRIMARY KEY,
+    raw_tx           BYTEA              NOT NULL,
+    contract_address TEXT               NOT NULL,
+    blob_sidecar     BYTEA,
+    tx_hash          TEXT,
+    tx_type          transaction_type   NOT NULL,
+    status           transaction_status NOT NULL,
+
+    created_at       TIMESTAMP          NOT NULL,
+    updated_at       TIMESTAMP          NOT NULL
+);
+
+-- for some time, we will need to fetch both old and new transactions
```

Comment:
This will be the first step of the migration. After I'm sure that on every env we have started using l1_transactions table, I will backfill l1_transactions table and stop using eth_txs table in core code.